src/api.jl

```diff
@@ -196,51 +408,78 @@ function infer(fieldids::Vector{Tuple{Int, Int, Int}},
     # initialize tiled images and model parameters for trimming.  We will
     # initialize the psf again before fitting, so we don't do it here.
     info("initializing celeste without PSF fit")
+    tic()
     tiled_images = SkyImages.break_blob_into_tiles(images, TILE_WIDTH)
     mp = ModelInit.initialize_model_params(tiled_images, images, catalog,
                                            fit_psf=false)
+    times.init_mp = toq()
 
     # get indicies of all sources relevant to those we're actually
     # interested in, and fit a local PSF for those sources (since we skipped
     # fitting the PSF for the whole catalog above)
     info("fitting PSF for all relevant sources")
+    tic()
     ModelInit.fit_object_psfs!(mp, target_sources, images)
+    times.fit_psf = toq()
+
+    # need a ModelParams per thread
+    mp_array = Array(ModelParams, nthreads())
```

Comment:
```ModelInit.get_relevant_sources()``` now makes a copy of ```mp```, so only ```mp_source``` is modified during optimization.  ```mp``` is never modified, so there's no need for this array anymore.